### ISSUE TITLE
[moe-quant-banks-e2e] Load and forward Qwen3.5-35B-A3B-GPTQ-Int4 within RAM budget

### ISSUE THREAD

**Issue by aangelinsf** [Performant-Labs/FreeToken-Intel#138]
## Slug
`moe-quant-banks-e2e`

## Why
The integration/validation step for #134: prove the packed-bank design
(#135, #136, #137) actually works against the real checkpoint, on real
hardware, within the real RAM budget -- not just against synthetic
fixtures. Sequential by nature (depends on #135, #136, #137 all merging
first); not parallel work.

## Fill in
No new production code expected (this issue is validation) -- may surface
small integration bugs in #135/#136/#137 to fix as they're found.

## Setup already done
- `Qwen/Qwen3.5-35B-A3B-GPTQ-Int4` (22.73 GiB, official Qwen release) is
  already downloaded to `/models/weights/qwen3.5-35b-a3b-gptq-int4` on
  `jupiter` -- no download needed to start this issue.
- The `qwen3_5_moe.parse_config` `model_path` bug that blocked ALL
  qwen3.5/3.6 checkpoint loading is already fixed (#131, merged).

## Accept
- `load_model("/models/weights/qwen3.5-35b-a3b-gptq-int4", moe_backend="offload")`
  completes without OOM, with real `free -h` / `dmesg` measurements recorded
  before and after (standing GPU-safety protocol: watch memory and exec-
  queue-reset counts, escalate load-only before attempting generation)
- Host RAM usage after load is close to the checkpoint's packed size
  (~23GB), NOT the ~88GB a full eager dequant would need -- record the real
  number as proof the packed-bank design worked
- A single forward step (prefill, a handful of tokens) succeeds and
  produces finite (non-NaN/non-inf) logits
- Generation quality (does the model produce coherent, correct text) is
  explicitly OUT of scope for this issue -- that is real further work once
  weight placement and a forward pass are proven correct

## Tests
Real XPU hardware, real checkpoint. Escalate carefully per the standing
GPU-safety protocol: confirm `dmesg`/`free -h` baseline before starting,
load-only first (no generation), re-check after. Do not batch this with
any other XPU-heavy test in the same session step.

Parent epic: #134
Related: #131, #135, #136, #137

**Comment by aangelinsf** [comment]
**Full accept criteria met, on real hardware, against the real checkpoint.**

```
load done in 205.5s
forward done in 8.6s
logits shape: torch.Size([1, 248320])
finite: True
FORWARD OK, peak RSS 21.66 GiB
```

`load_model("/models/weights/qwen3.5-35b-a3b-gptq-int4", moe_backend="offload")` builds a `quant_format="gptq_int4"` cache (256 experts, 40 layers, `gptq_group_size=128`, all matching the real checkpoint's config) and completes within **21.66GB peak RSS** -- not the ~88GB an eager bf16 dequant would have needed, the entire point of epic #134. A real forward step (prefill, 6 tokens) through the offload path produces finite `[1, vocab_size]` logits. `dmesg` exec-queue-reset count unchanged throughout (7, stable) and `free -h` showed healthy headroom at every measurement.

**This was not a clean first try -- #138 earned its keep as the sequential integration step.** #135/#136/#137 each passed their own synthetic-fixture tests in isolation, but wiring them together against the real 22.73GB checkpoint surfaced four real, independent bugs no synthetic fixture was large/real enough to catch:

1. **Loader never actually wired the three pieces together** (#144) -- `_attach_offload_cache` unconditionally built a `"bf16"`-schema cache with the wrong two bank names for every checkpoint; `load_moe_expert_sources` never dispatched to `stream_moe_expert_sources_gptq` at all. Also found and fixed: `iter_weights`' dense-placement pass was still eagerly dequantizing every routed expert's GPTQ tensors before discarding them -- the RAM blowup #135 was supposed to eliminate, reintroduced on a different call path.
2. **`stream_moe_expert_sources_gptq` buffered the whole checkpoint before finalizing anything** (#145) -- an earlier version held every layer's raw per-expert tensors until the entire stream ended, so peak RAM briefly held the raw checkpoint *and* the packed banks being built from it at once. Real, measured: exhausted a 27GB virtual-address ulimit on shard 6 of 14 before the fix. Fixed to finalize each `(layer, bank_name)` incrementally, releasing its raw buffer immediately.
3. **`iter_safetensors` opened (mmap'd) a shard file once per tensor, not once per shard** (#145) -- for the real checkpoint's per-expert layout (10,240 tensors in one ~1.4GB shard), this repeatedly mmap'd the same file thousands of times, fragmenting/exhausting virtual address space independent of real physical RAM need (confirmed: RSS always stayed far below what the ulimit failures implied). Fixed to group by shard and open each exactly once -- also a real ~30%+ load-time speedup.
4. **`SlotWeightAccessor` dequantized to the checkpoint's stored `scales` dtype (fp16) instead of the model's actual activation dtype (bf16)** (#146) -- crashed the very first real forward pass at the expert matmul with a dtype mismatch. No synthetic test caught it because every fixture happened to use one dtype throughout. Fixed to take an explicit `dtype` param, threaded from the activation tensor already in scope at both call sites.

Every fix shipped with its own regression test proving the specific property that broke (bit-exact dequant check, incremental-finalization call-order spy, shard-open-count spy, output-dtype-vs-scales-dtype check) -- not just re-running the same synthetic fixtures that already passed.

**Explicitly out of scope for this issue** (per its own accept criteria): generation quality (does the model produce coherent, correct text) -- weight placement and a forward pass are proven correct; whether the *outputs* are good is a separate, further step.

Closing #138. Real hardware, real checkpoint, real numbers.
